### ISSUE TITLE
Time queries with mobileToDesktop are non-idempotent

### ISSUE THREAD

**Issue by asapach** [browserslist/browserslist#804]
When running `browserslist` Node api multiple times using time queries, the `mobileToDesktop` option seems to mutate the data so that  the results are non-idempotent.

When first running `browserslist('last 1 year')` it returns the following list (notice only one `android` entry):
```
and_chr 119
and_ff 119
and_uc 15.5
android 119
chrome 119
...
```

However, once any `mobileToDesktop` query is executed it poisons the results:
```js
browserslist(null, { mobileToDesktop: true });
...
browserslist('last 1 year');
```
Notice the multiple `android` entries:
```
and_chr 119
and_ff 119
and_uc 15.5
android 119
android 118
android 117
android 116
android 115
android 114
android 113
android 112
android 111
android 110
android 109
chrome 119
...
```

`browserslist@4.22.1`
`caniuse-lite@1.0.30001565`

**Comment by ai** [comment]
Seems like we are not using immutable arrays somewhere.

Can you do the research and send PR with a fix?

**Comment by asapach** [comment]
> Can you do the research and send PR with a fix?

Here you go: #805

**Comment by ai** [comment]
The fix was released in 4.22.2.
